lib/dao.js

```diff
@@ -1748,7 +1763,14 @@ DataAccessObject.prototype.save = function (options, cb) {
 
     inst.isValid(function (valid) {
       if (valid) {
-        save();
+        Model.notifyObserversOf('persist', context, function(err) {
+          if (err) return cb(err);
+
+          data = inst.toObject(true);
+          Model.applyProperties(data, inst);
+          inst.setAttributes(data);
```

Comment:
I was required to add the following block of code in the `persist` hook, so that the new test `"applies updates from 'persist' hook"` passes successfully.

```
data = inst.toObject(true);
Model.applyProperties(data, inst);
inst.setAttributes(data);
```

But, after this block was added, another test `'should preserve properties with dynamic setters'` fails. The test `'should preserve properties with dynamic setters'` expects password to be `'bar-BAR'`, but owing to above block it gets `bar-BAR-BAR-BAR`

If we need to encrypt password within the `persist` hook, then, my block of code would be essential. But, I'm not sure, how to tackle the other test i.e. test `'should preserve properties with dynamic setters'`, in this case!